pandas/core/tools/datetimes.py

```diff
@@ -60,7 +74,42 @@ def _maybe_cache(arg, format, cache, convert_listlike):
     return cache_array
 
 
-def _convert_and_box_cache(arg, cache_array, box, errors, name=None):
+def _box_as_indexlike(
+    dt_array: ArrayLike,
+    tz: Optional[object] = None,
```

Comment:
Two general comments:

  1. If the default argument is `None` you don't need to type `Optional[...]` - mypy will infer that for you
  2. `object` is practically useless from a Typing perspective because it has almost no attributes or methods defined against it. `Any` is probably what you meant to use here, but if that's the case the annotation isn't doing anything (all objects without an annotation are referred to as `Any` by default)

Don't want this to be a hangup but if you remove `Optional` where not required here and other places should make things more readable. Also should remove object here and replace with something more useful, or just ignore for now

Reply:
In case you wanted to see point 1 in action, it's pretty easy to confirm with a sample like this:

```python
# in file foo.py
def foo(some_obj: str = None):
    reveal_type(some_obj)
    pass
```

```sh
$ mypy foo.py
foo.py:2: error: Revealed type is 'Union[builtins.str, None]'
```

Reply:
@WillAyd thanks!

I fixed it.